packages/codecs-core/src/fixCodec.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { fixBytes } from './bytes';
+import { Codec } from './codec';
+import { NotEnoughBytesError } from './errors';
+
+/**
+ * Creates a fixed-size codec from a given codec.
+ *
+ * @param codec - The codec to wrap into a fixed-size codec.
+ * @param fixedBytes - The fixed number of bytes to read.
+ * @param description - A custom description for the codec.
+ */
+export function fixCodec<T, U extends T = T>(
+    codec: Codec<T, U>,
+    fixedBytes: number,
+    description?: string
+): Codec<T, U> {
+    return {
+        description: description ?? `fixed(${fixedBytes}, ${codec.description})`,
+        fixedSize: fixedBytes,
+        maxSize: fixedBytes,
+        encode: (value: T) => fixBytes(codec.encode(value), fixedBytes),
+        decode: (buffer: Uint8Array, offset = 0) => {
+            // Slice the buffer to the fixed size.
+            buffer = buffer.slice(offset, offset + fixedBytes);
+            // Ensure we have enough bytes.
+            if (buffer.length < fixedBytes) {
+                throw new NotEnoughBytesError('fixCodec', fixedBytes, buffer.length);
```

Comment:
An example of my coded error comment above:

```ts
throw new WhateverWeCallThisError(
  'message',
  WhateverWeCallThisErrorCode.NOT_ENOUGH_BYTES,
  {codec: 'fixCodec', expected: fixedBytes, actual: bytes.length},
);

Reply:
Reverted to plain `Error`s for now with the `// TODO: Coded error.` prefix.